types/project.go

```diff
@@ -120,21 +121,21 @@ func (p *Project) ServicesWithBuild() []string {
 	servicesBuild := p.Services.Filter(func(s ServiceConfig) bool {
 		return s.Build != nil && s.Build.Context != ""
 	})
-	return maps.Keys(servicesBuild)
+	return slices.AppendSeq(make([]string, 0), maps.Keys(servicesBuild))
```

Comment:
`slices.Collect` will return nil which break testing.

Reply:
Why is `AppendSeq` required here, as maps.Keys already return a `[]string`?

Reply:
it returns iter.Seq[string]

Reply:
testing uses `assert.DeepEqual(t, []string{}, p.ServicesWithBuild())`, you can just switch to `assert.Equal(t, 0, len(p.ServicesWithBuild())` as nil and empty slice should be considered equivalent

Reply:
done in https://github.com/compose-spec/compose-go/pull/744

Reply:
not needed anymore AFAICT
otherwise, fix in test should be easy as demonstrated on https://github.com/compose-spec/compose-go/pull/744/files#diff-5fe049f199cfa53efadb44e5077186d86422bd7b585f333dae441b6c60f1eef2L312-R311